### PR TITLE
[#492][CRUD] Add the index view templates to the generator

### LINKED ISSUE #492: [CRUD] Add the index view templates to the generator
## Why

The default view generator in Rails does not include Bootstrap classes and has a poor layout. To make the generator usable out of the box, the Rails template must ship with custom view templates.

- This story only focuses on the `index` view.
- Ensure the template is using the SLIM engine

## Who Benefits?

Green field project teams.

### PR DESCRIPTION
**Pull request by khangbui22**
- Close #492 

## What happened 👀

- [x] Update Gemfile to use `slim-rails` instead of `slim` to be able to generate slim views with slim generator
- [x] Add `lib/template.rb` file to copy `lib/templates/slim/scaffold/**.tt` files to destination
- [x] Add `index.html.slim.tt` as the slim template for index view
- [x] Add tests  

## Insight 📝

> [!NOTE]
> Note that this PR is depended on the #483 [PR](https://github.com/nimblehq/rails-templates/pull/485) which handles the adding of CRUD add-on with slim to the template

## Proof Of Work 📹

https://github.com/nimblehq/rails-templates/assets/36690900/938f0496-b6c9-4eb7-94d2-a508c6a67428

### CODE REVIEW

File: .template/addons/crud/lib/template.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+def copy_tt_files
+  src_path = File.join(__dir__, 'templates')
+  dst_path = 'lib/templates'
+
+  Dir.glob("#{src_path}/**/*.tt").each do |file_source|
+    relative_path = file_source.gsub("#{src_path}/", '')
+
+    copy_file(file_source, File.join(dst_path, relative_path))
+  end
+end
```
Comment (malparty) [inline .template/addons/crud/lib/template.rb:23]: Can you try this? :) I'm not sure if that would work, but that would be more elegant 😇 

```suggestion
def copy_template_files
  ignore_tt do
    directory 'lib/templates', renderTemplates: false
  end
end
```

See how to create the `ignore_tt` method in: https://stackoverflow.com/a/74875526/1914241

=> The block `ignore_tt` seems easier to understand :)
``
    ignore_tt do
      directory(...)
    end
```

Comment (khangbui22) [reply .template/addons/crud/lib/template.rb:23]: I actually took the second approach idea below this one in the same post from him haha 😅 . I will give the first one a try to see how it goes. Thank you 💪

Comment (khangbui22) [reply .template/addons/crud/lib/template.rb]: It **WORKED,** thank you very much 🙇 

Fixed in https://github.com/nimblehq/rails-templates/pull/505/commits/b17ce32c7adcceb722c8be2beb744162885fd711

File: .template/addons/crud/lib/template.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  #       `when /#{TEMPLATE_EXTNAME}$/` condition and
+  #        fallback to default `copy_file`
+  Thor::TEMPLATE_EXTNAME.concat '_no_match' # => .tt_no_match
+  yield
+ensure
+  # NOTE: make sure to undo our dirty work after the block
+  Thor::TEMPLATE_EXTNAME.chomp! '_no_match' # => .tt
+end
+
+def directory_with_tt(...)
```
Comment (malparty) [inline .template/addons/crud/lib/template.rb:20]:
```suggestion
def directory_without_tt(...)
```

I think we don't really need this method 🤔

File: .template/addons/crud/lib/template.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+def ignore_tt
+  # NOTE: change template extension so it would skip
+  #       `when /#{TEMPLATE_EXTNAME}$/` condition and
+  #        fallback to default `copy_file`
+  Thor::TEMPLATE_EXTNAME.concat '_no_match' # => .tt_no_match
+  yield
+ensure
+  # NOTE: make sure to undo our dirty work after the block
+  Thor::TEMPLATE_EXTNAME.chomp! '_no_match' # => .tt
+end
```
Comment (malparty) [inline .template/addons/crud/lib/template.rb:18]: Should we move this method in a `ThorUtils` module in `.template/lib/thor_utils.rb`? 🤔

File: .template/addons/crud/lib/template.rb (hunk 2 above)
Comment (khangbui22) [reply .template/addons/crud/lib/template.rb]: Good catch 👍 . Fixed in https://github.com/nimblehq/rails-templates/pull/505/commits/45faaf9fe79aa156dfbdad89e2476fbacf27a406

File: .template/addons/crud/lib/template.rb (hunk 3 above)
Comment (khangbui22) [reply .template/addons/crud/lib/template.rb]: Good idea. Thank you for the suggestion 🙏 

Updated in https://github.com/nimblehq/rails-templates/pull/505/commits/8efd9539c3643a723b13e13c088f990e9ffa5152

File: .template/lib/thor_utils.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+    # NOTE: change template extension so it would skip
+    #       `when /#{TEMPLATE_EXTNAME}$/` condition and
+    #        fallback to default `copy_file`
+    Thor::TEMPLATE_EXTNAME.concat '_no_match' # => .tt_no_match
+    yield
+  ensure
+    # NOTE: make sure to undo our dirty work after the block
+    Thor::TEMPLATE_EXTNAME.chomp! '_no_match' # => .tt
+  end
+end
```
Comment (malparty) [inline .template/lib/thor_utils.rb]: missing empty ending line 😇

Comment (khangbui22) [reply .template/lib/thor_utils.rb]: 😅 I have added this new ending line in the latest changes already. Sorry about that 🤦

**Review by malparty — APPROVED** [review]
Okay, I finally found the time to test it 🚀 

image

**Review by sanG-github — APPROVED** [review]

**Pull request merged**